rotkehlchen/history/events.py

```diff
@@ -97,7 +98,12 @@ def _query_services_for_trades(self, filter_query: TradesFilterQuery) -> None:
 
         if location is not None:
             if location not in excluded_locations:
-                self.query_location_latest_trades(location=location, from_ts=from_ts, to_ts=to_ts)
+                self.query_location_latest_trades(
+                    location=location,
+                    ignored_names=ignored_names,
+                    from_ts=from_ts,
+                    to_ts=to_ts,
+                )
```

Comment:
This is not what I meant either. Is much easier than this.

```python
excluded_locations = defaultdict(set)
for exchange in self.db.get_settings(cursor).non_syncing_exchanges:
    excluded_instances[exchange.location].add(exchange.name)
```

and there you do

```suggestion
            if location is not None:
                self.query_location_latest_trades(
                    location=location,
                    excluded_instances=excluded_instances.get(location, {}),
                    from_ts=from_ts,
                    to_ts=to_ts,
                )
```